### DISCUSSION TITLE
Validating schema against empty json

### DISCUSSION THREAD

**Discussion by aij0** [python-jsonschema/jsonschema#937] [Q&A] [answered]
Hi,

I was wondering why doesn't the jsonschema.validate() raise an exception if I have empty json.
Am I missing some option or have I misunderstood the what the validation should check?

```
serverSchema = {
        "type": "object",
        "properties": {
            "server": {"type": "string"},
            "address": {"type": "string", "format": "ip-address"},
            "poll_period": {"type": "number"},
            "regex": {"type": "string"}
        }
    }
```
If I test this against `server = {}` it passes without exceptions.
Also if I give it a nested json it also passes without any exceptions:
```
        nested_json = {
            "servers": [
                {"server": 1234,
                    "address": "hello",
                    "poll_period": "there",
                    "regex": "Weight"
                 },
                {"server": "fourth",
                    "address": "127.0.0.1:5000",
                    "poll_period": 120,
                    "regex": ""
                 }
            ]
        }
```

Finally if I have the following json which is like the schema but with wrong types it will raise an Exception as expected.

```
        invalid_json = {
                {"server": 1234,
                    "address": "hello",
                    "poll_period": "there",
                    "regex": "Weight"
                 }
```

**Comment by Julian** [comment, marked as answer]
In the JSON Schema specification (all current versions of it), properties are not required by default until you say so, either via the `required` validator or by saying `additionalProperties: false`.

Without doing so, what your schema means is simply that those subschemas apply only *if* the corresponding property is present.

**Reply by aij0** [reply]
Thank you for quick response!
